src/cupbearer/scripts/train_detector.py

```diff
@@ -6,7 +6,10 @@
 
 
 def main(cfg: Config):
-    reference_data = cfg.task.build_train_data()
+    reference_data = cfg.task.build_train_data(
+        # TODO perhaps have the 0.1 to be configurable as well
```

Comment:
I think a default of e.g. 0.9 seems more appropriate, since in backdoor detection, you'd usually assume you mostly have clean data, with some (<=10%) poisoned samples, so I think 90%+ clean makes sense here.

Ultimately, this feels like a slightly wrong abstraction anyway, realistically the *task* should determine the ratio (and also whether non-clean data is even available for training). But I'd be fine with changing that later, there might be other changes we need to make anyway as we integrate more tasks

Reply:
Oh, thanks for the catch. For theory to hold needs to be in (1.0, 0.5), a mistranslation from poisoning ratio.

 I agree that the poisoning ratio should belong to the task. I'm unsure about whether non-clean data should be available for training. Whether a detector can be trained on non-clean data or not is a property of the detector.
 
 Hmm, but yes I think I just convinced myself that you're right. I'll do something like:
  - [ ] Everything about build_train_data belongs to task
  - [ ] Class attributes for detectors to check whether they can be trained on poisoned/clean
  - [ ] Warning if usage is incompatible